### ISSUE TITLE
to add a feasible nodes/failed no cache for same type of pods in same workload

### ISSUE THREAD

**Issue by lowang-bh** [kubernetes/kubernetes#124949]
### What would you like to be added?

A cached feasible nodes list for a worknode, eg, deployment.

### Why is this needed?

As predicating is a time consume step, especially in a large cluster.
And as we know, same type of pods in a workload have same constraint, such as deployment's pod, Job's replicas.
So the first pod's feasible nodes list is aslo avaiable for the other pods in a workload.
Choose one from the cached nodes will save much time than re-run a predicating for next pod.

The same does not a  node list predicating failed.

**Comment by zhifei92** [comment]
Which component do you want to implement this functionality in, is it the schdeuler? And If podAffinity is present, the distribution of pods in the deployment will be affected.

**Comment by chengjoey** [comment]
even through same type of pods in same workload, if set `volume` or `TopologySpreadConstraint`, feasible nodes should be a big difference

**Comment by AxeZhan** [comment]
/sig scheduling
Sorry, I didn't look into the details. As your purpose is to accelerating predicate, this belongs to sig-scheduling.

I think I can understand some of what you're trying to do.
For example, if we have a deployment with required nodeAffinity.
Then maybe we can store a affinity->node mapping, and then we can skip `preFilter` of `nodeAffinity` plugin for other pods in the same deployment.

However, the maintaining of this cache will also increase the burden on the scheduler, and brings additional complexity to the framework.
If you have implemented this somewhere, then maybe the results of a benchmark will be very convincing.

**Comment by lowang-bh** [comment]
> And If podAffinity is present, the distribution of pods in the deployment will be affected.

all pods in  a deployment have same predicate condition, include podAffinity, right？

**Comment by lowang-bh** [comment]
> plugins such as `resource`, `affinity`, feasible nodes should be a big difference

`Resource` should be re-considered, of couse.

**Comment by lowang-bh** [comment]
> However, the maintaining of this cache will also increase the burden on the scheduler, and brings additional complexity to the framework. If you have implemented this somewhere, then maybe the results of a benchmark will be very convincing.

Yes, it is a heavy work. I just throw out this question and we can discuss it.  Some other schedulers based on kubernetes have this feature.

**Comment by AxeZhan** [comment]
> Some other schedulers based on kubernetes have this feature.

Can you paste a reference here? :)

**Comment by basuotian** [comment]
Does this feature look like `equivelance class cache` which is removed in v1.23?

**Comment by lowang-bh** [comment]
> > Some other schedulers based on kubernetes have this feature.
> 
> Can you paste a reference here? :)

such as the godel-scheduler in bytedance's recent paper described.
and volcano has add it in https://github.com/volcano-sh/volcano/pull/1165.

https://github.com/volcano-sh/volcano/blob/70a483b66653ee402babdc733505a92fff890f5f/pkg/scheduler/plugins/predicates/predicates.go#L434

**Comment by lowang-bh** [comment]
> equivelance class cache

Yes, seem it has been discussed in 
https://github.com/kubernetes/kubernetes/issues/17390,
https://github.com/kubernetes/kubernetes/issues/71013
https://github.com/kubernetes/kubernetes/pull/65714

removed in https://github.com/kubernetes/kubernetes/pull/71399
